_pytest/main.py

```diff
@@ -210,7 +210,9 @@ def __getitem__(self, key):
             return self.parent.keywords[key]
 
     def __setitem__(self, key, value):
-        self._markers[key] = value
+        from _pytest.mark import MarkDecorator
+        if not isinstance(self._markers.get(key), MarkDecorator):
```

Comment:
I tested the patch locally and it works as advertised, I'm just not so sure if the fix is correct. @RonnyPfannschmidt has more experience with marks in general, what do you think?